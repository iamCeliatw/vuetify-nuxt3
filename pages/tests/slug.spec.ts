import { mount } from '@vue/test-utils';
import { describe, it, expect, vi, beforeEach, type Mock } from 'vitest';
import Slug from '../[...slug].vue';
import { useRoute } from 'vue-router';

// 模拟 definePageMeta 函数
vi.stubGlobal('definePageMeta', vi.fn());
vi.stubGlobal('setPageLayout', vi.fn());
const mockUseRoute = () => {
  //params 
  vi.stubGlobal('useRoute', vi.fn(() => ({
    params: {
      slug: ['admin', 'path']
    }
  }))
);
}
// 模拟 vue-router 模块
vi.mock('vue-router', () => ({
  useRoute: vi.fn()
}));

describe('Slug', () => {
  beforeEach(() => {
    mockUseRoute();
    const routeMock = {
      params: {
        slug: ['admin', 'path']
      }
    };
    (useRoute as unknown as Mock).mockReturnValue(routeMock);
  });

  it('renders correctly', () => {
    const wrapper = mount(Slug);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
