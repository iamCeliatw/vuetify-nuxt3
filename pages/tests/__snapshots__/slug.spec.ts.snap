// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slug > renders correctly 1`] = `
"<nuxtlayout data-v-575d0b4c="">
  <div data-v-575d0b4c="" class="noPage__wrapper">
    <div data-v-575d0b4c="" class="noPage__container">
      <h1 data-v-575d0b4c="">404</h1>
    </div>
    <div data-v-575d0b4c="" class="planet"><img data-v-575d0b4c="" src="/planet.png"></div>
    <div data-v-575d0b4c="" class="alien"><img data-v-575d0b4c="" src="/alien.png"></div>
    <div data-v-575d0b4c="" class="not_found">
      <div data-v-575d0b4c="" class="text">
        <h1 data-v-575d0b4c="" class="title">404</h1>
        <p data-v-575d0b4c="">page not found</p>
      </div>
      <div data-v-575d0b4c="" class="back_page"><img data-v-575d0b4c="" src="/arrow-back.png">
        <router-link data-v-575d0b4c="" to="/">Go to homepage</router-link>
      </div>
    </div>
  </div>
</nuxtlayout>"
`;
